cranelift/codegen/src/isa/zkasm/abi.rs

```diff
@@ -346,14 +346,14 @@ impl ABIMachineSpec for ZkAsmMachineDeps {
         let mut insts = SmallVec::new();
 
         if frame_layout.setup_area_size > 0 {
+            insts.push(Inst::ReserveSp {
+                amount: frame_layout.setup_area_size.into(),
+            });
             insts.push(Self::gen_store_stack(
                 StackAMode::SPOffset(0, I64),
```

Comment:
Good point. I think we can actually simplify the setup, given that we don't rely on frame pointer register. We only need to preserve the `RR` for non-leaf functions and otherwise can skip the frame setup altogether.
I've updated the code to do this, PTAL.

Reply:
I am a little confused as to why you've chosen to go with assertions here but code seems alright now. And yeah I dont think we want to make an already tricky PR any trickier.

Reply:
Do you mean adjusting to something like `SPOffset(setup_area_size - 8, I64)`? Mainly because it is harder to read than the special case version that we have.